### PR TITLE
boards/common/nucleo: Marked board_init as weak

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Made `void board_init(void)` in the common implementation a weak symbol to allow providing a custom board init, if the common nucleo board init misses features.

The testbed nodes developed here at the University in Magdeburg ([presented at the RIOT summit](https://www.youtube.com/watch?v=kcJnExd2NHQ)) are based on a Nucleo, but require a custom init routine for the initialization of the attached hardware.

### Testing procedure

Initialization on the nucleos should work. In fact, the generated binaries should not change.

### Issues/PRs references

None

### CODE REVIEW

**Comment by cladmi** [comment]
I will do a way longer comment than needed for this one but I think it is necessary.
TL;DR: with a few changes it will be good, but I do not think it is a long term solution in that form.

File: boards/common/nucleo/board.c [hunk 1]
```diff
@@ -31,7 +31,12 @@
 #include "board.h"
 #include "periph/gpio.h"
 
-void board_init(void)
+/*
+ * Allow overwriting board_init if common implementation doesn't work.
+ * If at link time another implementation of board_init() not marked as weak
+ * ((a.k.a. a strong symbol) is present, it will be linked in instead.
+ */
+void __attribute__((weak)) board_init(void)
```
[the remark this answers is not in the document]
Comment (maribu) [reply boards/common/nucleo/board.c]: Done

**Comment by benpicco** [comment]
A more general solution would be nice, but the change itself is small and shouldn't be a problem.
If it's still useful for you, please squash!

**Comment by fjmolinas** [comment]
@benpicco only missing your ACK here ;)

**Comment by benpicco** [comment]
@fjmolinas I think your ACK would have worked as well :wink:

**Comment by maribu** [comment]
All green :-)
